UsbIpServer/CommandHandlers.cs

```diff
@@ -416,6 +416,31 @@ async Task<ExitCode> ICommandHandlers.WslAttach(BusId busId, string? distributio
                 }
             }
 
+            // 8) Heuristical firewall check
+            //
+            // With minimal requirements (bash only) try to connect from WSL to our server.
+            // If the process does not terminate within one second, then most likely a third party
+            // firewall is blocking the connection. Anything else (e.g. bash not available, or not supporting
+            // /dev/tcp, or whatever) will most likely finish within 1 second and the test will simply pass.
+            // In any case, just issue a warning, which is a lot more informative than the 1 minute TCP
+            // timeout that usbip will get.
+            {
+                using var timeoutTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(1));
+                using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, timeoutTokenSource.Token);
+                try
+                {
+                    var wslResult = await ProcessUtils.RunCapturedProcessAsync(
+                        WslDistributions.WslPath,
+                        (distribution is not null ? new[] { "--distribution", distribution } : Enumerable.Empty<string>()).Concat(
+                            new[] { "--user", "root", "--", "bash", "-c", $"echo < /dev/tcp/{distros.HostAddress}/{Interop.UsbIp.USBIP_PORT}" }),
+                        Encoding.UTF8, linkedTokenSource.Token);
+                }
```

Comment:
Should we also check the error code? I'm imagining a situation where the user doesn't have bash installed. In that case, a more generic warning message followed by any captured output might be more helpful.

Reply:
I thought about that. But that would produce a warning like "bash not available" or "/dev/tcp not found". Not something the user can easily fix. And it would most likely be a non-issue, as this check is only for a third-party firewall; most users would pass the test anyway. So, if the user has no bash, or a weird version of bash, and (most likely) do not have a third-party firewall, then no harm is done by *not* giving a warning that the test couldn't be executed.

The only false results remaining is: user has a third-party firewall *and* did not open port 3240 (as instructed in the documentation) *and* has some weird distro without bash. In that case they do not get this early warning, but a "cannot connect" from usbip itself after about a minute, just as in previous versions. I can live with that...